CMakeLists.txt

```diff
@@ -142,22 +142,30 @@ if(NOT DEFINED SNMALLOC_ONLY_HEADER_LIBRARY)
     set(${result} ${dirlist})
   endmacro()
 
-  macro(add_shim name)
-    if(SNMALLOC_RUST_SUPPORT)
-      add_library(${name} STATIC src/override/rust.cc)
+  macro(add_shim name type)
+    add_library(${name} ${type} ${ARGN})
+    target_link_libraries(${name} snmalloc_lib)
+    if(MSVC)
+      target_compile_definitions(${name} PRIVATE "SNMALLOC_EXPORT=__declspec(dllexport)")
```

Comment:
Perhaps this is not actually needed, but have it added anyway.

Reply:
I am not sure what is required to link this into Rust. As this is a static lib, I would have said it isn't needed. @snf any thoughts?

Reply:
Pretty sure it's not required for a static lib

Reply:
Could you drop this bit please, and I'll merge.

Reply:
@mjp41 I went to beg by then. Sorry for the late reply.